staging/src/k8s.io/component-base/logs/datapol/externaltypes.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datapol
+
+import (
+	"fmt"
+	"reflect"
+)
+
+const (
+	vendorPrefix = "k8s.io/kubernetes/vendor/"
+
+	oauth2Token           = "golang.org/x/oauth2.Token"
+	jwtJSONWebToken       = "gopkg.in/square/go-jose.v2/jwt.JSONWebToken"
+	jwtNestedJSONWebToken = "gopkg.in/square/go-jose.v2/jwt.NestedJSONWebToken"
+)
+
+// GlobalDatapolicyMapping returns the list of sensitive datatypes are embedded
+// in types not native to Kubernetes.
+func GlobalDatapolicyMapping(v interface{}) []string {
+	return byType(reflect.TypeOf(v))
+}
+
+func byType(t reflect.Type) []string {
+	// Use string representation of the type to prevent taking a depency on the actual type.
+	switch fmt.Sprintf("%s.%s", t.PkgPath(), t.Name()) {
+	case "net/http.Header":
+		return []string{"password", "token"}
+	case "net/http.Cookie":
+		return []string{"token"}
+	case "crypto/x509.Certificate":
+		return []string{"security-key"}
+	case oauth2Token, vendorPrefix + oauth2Token:
+		return []string{"token"}
+	case jwtJSONWebToken, vendorPrefix + jwtJSONWebToken:
+		return []string{"token"}
+	case jwtNestedJSONWebToken, vendorPrefix + jwtNestedJSONWebToken:
+		return []string{"token"}
+	default:
```

Comment:
Consider extracting the first cases to `const` for consistency with the latter three cases.

Reply:
done